### PR TITLE
Skip JumpList on NanoServer and IoT

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

The STA apartment state is not supported on NanoServer and Windows IoT.
Skip JumpList creation on NanoServer and IoT. 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Review by adityapatwardhan — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

File: .vsts-ci/templates/nanoserver.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+parameters:
```
Comment (PoshChan) [inline .vsts-ci/templates/nanoserver.yml]: @TravisEz13, your last commit had 1 failures in `PowerShell-CI-windows`
CmsMessage cmdlets and Get-PfxCertificate basic tests.Verify a CMS message can be protected / unprotected
```powershell
Keyset does not exist
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Security\CmsMessage.Tests.ps1: line 71
71:         $decrypted = $message | Unprotect-CmsMessage -To $certLocation

```

File: .vsts-ci/templates/nanoserver.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+jobs:
+
+- job: ${{ parameters.jobName }}
+  variables:
+    scriptName: ${{ parameters.scriptName }}
+
+  pool:
+    vmImage: ${{ parameters.vmImage }}
+  #container: ${{ parameters.container }}
```
Comment (adityapatwardhan) [inline .vsts-ci/templates/nanoserver.yml]: nit: remove this

File: .vsts-ci/templates/nanoserver.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+parameters:
+  vmImage: 'win1803'
+  jobName: 'Nanoserver_Tests'
+  container: 'mcr.microsoft.com/powershell:nanoserver-1803'
```
Comment (adityapatwardhan) [inline .vsts-ci/templates/nanoserver.yml]: Remove this

**Review by adityapatwardhan — APPROVED** [review]

File: .vsts-ci/templates/nanoserver.yml (hunk 2 above)
Comment (TravisEz13) [reply .vsts-ci/templates/nanoserver.yml]:
```suggestion
```

File: .vsts-ci/templates/nanoserver.yml (hunk 3 above)
Comment (TravisEz13) [reply .vsts-ci/templates/nanoserver.yml]:
```suggestion
```

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 20 minutes

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 20 minutes ago

File: test/nanoserver/nanoserver.tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Describe "Verify PowerShell Runs" {
```
Comment (iSazonov) [inline test/nanoserver/nanoserver.tests.ps1]: License header?

Comment (TravisEz13) [reply test/nanoserver/nanoserver.tests.ps1]: @daxian-dbw  or @iSazonov  can you add this?

Comment (daxian-dbw) [reply test/nanoserver/nanoserver.tests.ps1]: Submitted #10171

**Pull request merged**